crates/host/src/wasmbus/mod.rs

```diff
@@ -1977,14 +1982,34 @@ impl Host {
 
         let annotations = annotations.map(|annotations| annotations.into_iter().collect());
         let mut providers = self.providers.write().await;
-        let hash_map::Entry::Occupied(mut entry) = providers.entry(provider_ref) else {
+        let hash_map::Entry::Occupied(mut entry) = providers.entry(provider_ref.clone()) else {
             return Ok(SUCCESS.into());
         };
         let provider = entry.get_mut();
         let instances = &mut provider.instances;
         if let hash_map::Entry::Occupied(entry) = instances.entry(link_name.clone()) {
             if entry.get().annotations == annotations {
                 let ProviderInstance { id, child, .. } = entry.remove();
+
+                // Send a request to the provider, requesting a graceful shutdown
+                if let Ok(payload) = serde_json::to_vec(&json!({ "host_id": host_id })) {
```

Comment:
Since we know the contents of this payload and it's simple, we should probably assert we can serialize and bail if we can't

Reply:
I chose to do it this way because even if we were to fail to send the message or serialize (which should be impossible) we definitely want to proceed to aborting the child process. For that reason I don't think we should bail, and this destructuring here is really just out of an abundance of caution

Reply:
Fair enough